### PR TITLE
Pytest extension

### PR DESCRIPTION
**Pull request by maurelius167**
Adds a small section about pytest.

Addresses #81. I am looking for feedback before #87 is addressed.

### CODE REVIEW

**Comment by john-science** [comment]
Awesome!

I didn't find this until too late in the day, but I will take a look in the morning. 

Thanks!

File: classes/17_testing_projects/lecture_17.md [hunk 1]
```diff
@@ -4,7 +4,10 @@ Today we're going to begin the big conversation of testing code. This lecture do
 
 #### Installation
 
-None! Everything we talk about today is part of the Python standard library.
+In today's lecture we will cover two testing frameworks: `unittest` and `pytest`.
+The `unittest` framework is included with python, and no additional work is required if python is already installed.
+The `pytest` framework is widely availble with `pip`, `conda`, or virtually any other python package manager.
```
Comment (john-science) [inline classes/17_testing_projects/lecture_17.md]:
```suggestion
The `unittest` framework is included with Python, and no additional work is required if Python is already installed.
The `pytest` framework is widely available with `pip`, `conda`, or virtually any other Python package manager.
```

File: classes/17_testing_projects/lecture_17.md [hunk 2]
```diff
@@ -404,6 +411,29 @@ First, you can ignore the `EX_HTML` above, we just hard code some test/sample/ex
 
 More importantly, see that we have used `unittest.mock.patch` to NOT really call `requests.get()` but replace that code with a `MagicMock` that returns the data we want. This is the crux we need to understand to build fast, maintainabe tests. And mocking allows us to easily write tests for code that would otherwise _seem_ untestable.
 
+## Pytest
+
+[Pytest](https://docs.pytest.org/en/8.0.x/) is a popular alternative to unittest.
+One nice feature is that tests that have been written with the unittest framework can already be run with pytest.
+For example, the file `test_student.py` can be run on the command line as follows:
```
Comment (john-science) [inline classes/17_testing_projects/lecture_17.md]:
```suggestion
For example, the file `test_student.py` above can be run on the command line as follows:
```

File: classes/17_testing_projects/lecture_17.md [hunk 3]
```diff
@@ -404,6 +411,29 @@ First, you can ignore the `EX_HTML` above, we just hard code some test/sample/ex
...
+    platform <platform info>
+    rootdir: <path to rootdir>
+    collected 3 items
+
+    test_student.py ...                                                      [100%]
+
+    ============================== 3 passed in 0.00s ===============================
+
+
+Some advantages that pytest has over unittest is that it enables more pythonic ways to write unit tests, and it has many [helpful plugins](https://docs.pytest.org/en/8.0.x/reference/plugin_list.html#plugin-list).
```
Comment (john-science) [inline classes/17_testing_projects/lecture_17.md]:
```suggestion
One advantage of `pytest` over `unittest` is that it supports [plugins](https://docs.pytest.org/en/8.0.x/reference/plugin_list.html#plugin-list).
```

File: classes/17_testing_projects/lecture_17.md [hunk 4]
```diff
@@ -404,6 +411,29 @@ First, you can ignore the `EX_HTML` above, we just hard code some test/sample/ex
...
+There are two python projects worth mentioning that are compatible with pytest are [hypothesis](https://hypothesis.readthedocs.io/en/latest/) and [pytest-dependency](https://pypi.org/project/pytest-dependency/).
```
Comment (john-science) [inline classes/17_testing_projects/lecture_17.md]:
```suggestion
There are two Python projects worth mentioning that are compatible with pytest: [hypothesis](https://hypothesis.readthedocs.io/en/latest/) and [pytest-dependency](https://pypi.org/project/pytest-dependency/).
```

File: classes/17_testing_projects/lecture_17.md [hunk 5]
```diff
@@ -404,6 +411,29 @@ First, you can ignore the `EX_HTML` above, we just hard code some test/sample/ex
...
+
+For more information on how to get started with pytest checkout the [Get Started](https://docs.pytest.org/en/8.0.x/getting-started.html#get-started) page for a basic introduction.
+For a more indepth introduction [this two hour tutorial](https://www.youtube.com/watch?v=LX2ksGYXJ80) covers pytest in scientific computing applications.
```
Comment (john-science) [inline classes/17_testing_projects/lecture_17.md]:
```suggestion
For a more in-depth introduction [this two hour tutorial](https://www.youtube.com/watch?v=LX2ksGYXJ80) covers pytest in scientific computing applications.
```

**Comment by john-science** [comment]
@maurelius167 Love it!  

Take a look at my comments.  They are all/mostly just grammar and capitalization fixes; nothing major.

If you like them, accept them, and we'll merge!

(Sorry for the long delay, I got busy.)

**Review by maurelius167 — COMMENTED** [review]
Thanks @john-science, and sorry for the delayed response as well. I try to get to things as soon as I can, but life 🤷 You're changes look great, much appreciated!

I wanted to keep pytest extension short for this first iteration because there are already A LOT of great resources for pytest. No need to duplicate tutorials.
 
Let me know if there is anything you'd like to know more about. Recently, I've done some deep dives into the pytest ecosystem. I can provide more references or expand on the examples in whatever direction.

**Review by john-science — APPROVED** [review]

**Pull request merged**